packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -2896,10 +2896,14 @@ enum _RouteLifecycle {
 
 typedef _RouteEntryPredicate = bool Function(_RouteEntry entry);
 
-class _NotAnnounced extends Route<void> {
-  // A placeholder for the lastAnnouncedPreviousRoute, the
-  // lastAnnouncedPoppedNextRoute, and the lastAnnouncedNextRoute before any
-  // change has been announced.
+/// Placeholder for a route.
+abstract class _RoutePlaceholder {
+  const _RoutePlaceholder();
+}
+
+/// Not announced route.
+class _NotAnnounced extends _RoutePlaceholder {
+  const _NotAnnounced();
 }
```

Comment:
What do we need the extra level of inheritance for? Could we just make _RoutePlaceholder non-abstract and use that?

Reply:
There are already two options for not existing route: null and not-announced. If we ever get third one, we will not be able to use const _RoutePlaceholder(), because it is already occupied So we will need to introduce more const constructors or extra level of hierarchy. But, yes, it can be postponed.